events.c

```diff
@@ -434,6 +463,76 @@ static int EVT_clean_fdsets(struct EventState *ctx)
    return foundOne;
 }
 
+char EVT_enable_virt(EVTHandler *ctx, struct timeval initTime)
+{
+   struct VirtClkState *sim;
+
+   if (!ctx) {
+      return -1;
+   }
+
+   global_evt = ctx;
+   
+   sim = CLK_init(&initTime);
+   if (!sim) {
+      return -1;
+   }
+   
+   ctx->enableVirtClk = CLK_ENABLED;
+   ctx->virtClkState = sim;
+   return 0;
+}
+
+struct VirtClkState *EVT_clk(EVTHandler *ctx)
+{
+   return ctx->virtClkState;
+}
+
+/**
+ * Get the system's current absolute GMT time.
+ *
+ * @param tv A pointer to the timeval structure where the time gets stored
+ *
+ */
+int EVT_get_gmt_time(EVTHandler *ctx, struct timeval *tv)
+{
+   if (ctx && ctx->enableVirtClk) {
+      // Handle simulation
+      *tv = CLK_get_time(ctx->virtClkState);
+   } else {
+      gettimeofday(tv, NULL);
+   }
+
+   return 0;
+}
+
+int EVT_get_gmt_time_virt(struct timeval *tv)
+{
+   if (global_evt && global_evt->enableVirtClk) {
+      // Handle simulation
+      *tv = CLK_get_time(global_evt->virtClkState);
+      return 0;
+   }
+
+   return -1;
+}
+
+#ifndef __APPLE__
+int EVT_get_monotonic_time(EVTHandler *ctx, struct timeval *tv)
```

Comment:
Why doesn't monotonic time try to handle virtual time correctly? It seems confusing to wrap getting the time with `get_time` that decides weather or not to use gmt or monotonic time.

Instead, could we do that logic inside on EVT_get_monotonic time, so that it "just works" when simulating virtual time?